research-software-directory/RSD-as-a-service
# Add more info to software licenses

# Software licenses

Closes #1033 
Closes #1041

Changes proposed in this pull request:
* Save more information about software license (name, reference and open_source flag)
* Refactor: move configuration of links & metadata section to separate config file 
* Chore: fix found type errors
* Fix failing tests 
* Use short identifier as main label and show the long license name on mouseover. We also provide the link to the license page (by default spdx page).
* Search for license by short identifier or by long name
* For custom licenses providing reference link (url to license page) is required

How to test:
* `make start` to build and create test data
* Login and create new software
* Go to links & metadata section and add licenses:
  * Type Apache and select license
  * Try import using concept DOI 10.5281/zenodo.6379973
  * Type unknown license name like `My new fancy license` and add custom license
* Preview software page, confirm the licenses are shown and the links are working

## TODO!
** Create migration script to update all historical data for the spdx license entries **

## Edit software license page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/b7c940ac-d438-4c30-a828-bb8c95814287)

## Add custom license
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/636c9f39-30d7-44d1-8e9e-e592c247946c)

## Software licenses on software page
![image](https://github.com/research-software-directory/RSD-as-a-service/assets/9204081/fd2b9cee-c608-4080-9d52-37dc9e92e85b)

PR Checklist:
